src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -427,7 +427,23 @@ internal void CopyFrom(ResourceDictionary source)
 			// In order to ensure the theme updates covers all of them, we need to keep track of this source instance for theme updates before it is lost.
 			_sourceDictionary = WeakReferencePool.RentWeakReference(this, source);
 
-			_values.AddRange(source._values);
+			// Lazy resource initialization needs the current XamlScope
+			// to resolve values, and the originally defined scope that 
+			// was set when the source dictionary was created may not be 
+			// value for the current XAML scope. We rewrite the initializer
+			// in order for the name resolution to work properly.
+			foreach (var pair in source._values)
```

Comment:
Consider calling _values.EnsureCapacity to make sure the repeated Add calls can't cause multiple resizes. This is something was done when using AddRange.

Not very important though, but makes sure we are not regressing performance in this code path

Reply:
Good catch! Thanks!